### PR TITLE
MessagePaginationAction and some adjustments to MessageHistory

### PR DESCRIPTION
**Pull request by MinnDevelopment**

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/core/events/guild/voice/GuildVoiceJoinEvent.java [hunk 1]
```diff
@@ -20,18 +20,12 @@
 import net.dv8tion.jda.core.entities.Member;
 import net.dv8tion.jda.core.entities.VoiceChannel;
 
-public class GuildVoiceJoinEvent extends GenericGuildVoiceEvent
+public class GuildVoiceJoinEvent extends GuildVoiceMoveEvent
```
Comment (Almighty-Alpaca) [inline src/main/java/net/dv8tion/jda/core/events/guild/voice/GuildVoiceJoinEvent.java]: I don't really like this, join != move

File: src/main/java/net/dv8tion/jda/core/events/guild/voice/GuildVoiceLeaveEvent.java [hunk 2]
```diff
@@ -20,18 +20,10 @@
 import net.dv8tion.jda.core.entities.Member;
 import net.dv8tion.jda.core.entities.VoiceChannel;
 
-public class GuildVoiceLeaveEvent extends GenericGuildVoiceEvent
+public class GuildVoiceLeaveEvent extends GuildVoiceMoveEvent
```
Comment (Almighty-Alpaca) [inline src/main/java/net/dv8tion/jda/core/events/guild/voice/GuildVoiceLeaveEvent.java]: same as above

**Comment by MinnDevelopment** [comment]
I'm not sure I like combining the voice events, either

**Comment by MinnDevelopment** [comment]
Changes made in this PR related to the PaginationIterator cache need to be squashed before audit log development can continue.

**Pull request merged**